design/simple-global-load-balancing.md

```diff
@@ -0,0 +1,938 @@
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1 -- DNS driven allocation
+
+In this example we assume taht the DNS entry for the globally load balanced service has been created by some extewrnal process, at this point we can proceed with the following defintions:
+
+cluster1:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+ name: my-service
+ annotations:
+   metallb.io/assign-from-fqdn: myglobal.service.example.io
+spec:
+ selector:
+   app.kubernetes.io/name: MyApp
+ ports:
+   - protocol: TCP
+     port: 80
+     targetPort: 9376
+ type: LoadBalancer
+```
+
+cluster2:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+ name: my-service
+ annotations:
+   metallb.io/assign-from-fqdn: myglobal.service.example.io
+spec:
+ selector:
+   app.kubernetes.io/name: MyApp
+ ports:
+   - protocol: TCP
+     port: 80
+     targetPort: 9376
+ type: LoadBalancer
+```
+
+both service's VIPs are assigned by resolving the FQDN from the DNS. They will have the same VIP, implementing the BGP-based global load balancer.
+
+#### Story 2 - external-DNS driver allocation
+
+In this use story, we assume that th external-DNS operator is available and correctly configured. Then we can do the following:
+
+cluster1:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+  annotations:
+    external-dns.alpha.kubernetes.io/hostname: myglobal.service.example.io
+spec:
+  selector:
+    app.kubernetes.io/name: MyApp
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 9376
+  type: LoadBalancer
+```
+
+cluster2
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+ name: my-service
+ annotations:
+   metallb.io/assign-from-fqdn: myglobal.service.example.io
+spec:
+ selector:
+   app.kubernetes.io/name: MyApp
+ ports:
+   - protocol: TCP
+     port: 80
+     targetPort: 9376
+ type: LoadBalancer
+```
+
+the service in cluster 1 is assigned by metallb with the usual means. Then external-dns creates the DNS entry in the managed DNS. After that the service in cluster 2 can resolve the fqdn and assigns the VIP, which is going to be the same as for cluster1.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+VIPs assigned via DNS will still belong to an address pool. A series of check will occur before the IP returned by the DNS will be actually assigned. At a minium the following checks:
+- the IP belongs to allowed CIDRs
+- the IP is available.
+
+A possible best practice to mitigate the risk of failure of these checks is to dedicate an IPaddress pool just ofr the globally load balanced services. This way the pool usage (taken IPs vs free IPs) will be the same across the clusters.
```

Comment:
This must be considered. In the version I am proposing (the ips go to the status sub structure), we won't notice conflicts until they happen. What I think it makes sense is to give IPs assigned manually precedence (i.e. if we have two IPAddressPools, where the IP belongs to one because of a manually assigned cidr, and to the other because of a dns source, the manual one will take precedence and we'll ignore the dns one).